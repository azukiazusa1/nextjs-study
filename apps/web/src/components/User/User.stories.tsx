import { ComponentMeta, ComponentStory } from '@storybook/react';
import React from 'react';

import User from './User';

export default {
  component: User,
} as ComponentMeta<typeof User>;

const Template: ComponentStory<typeof User> = (args) => <User {...args} />;

export const Default = Template.bind({});
Default.args = {
  username: '@johndoe',
  image: '/images/1.png',
  status: 'online',
  score: 12,
};

export const isOffline = Template.bind({});
isOffline.args = {
  username: '@johndoe',
  image: '/images/1.png',
  status: 'offline',
  score: 12,
};

export const hasMessage = Template.bind({});
hasMessage.args = {
  username: '@johndoe',
  image: '/images/1.png',
  status: 'online',
  score: 12,
  message: 'üëç',
};
