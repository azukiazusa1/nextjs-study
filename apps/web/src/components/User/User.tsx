import React from 'react';

import Avatar from '@/components/Avatar/Avatar';

type Props = {
  className?: string;
  username: string;
  image: string;
  status: 'online' | 'offline' | '';
  score: number;
  message?: string;
};

const User: React.FC<Props> = ({
  className = '',
  username,
  image,
  status,
  score,
  message = '',
}: Props) => {
  return (
    <div className={`flex max-w-60 ${className}`}>
      <div className={`tooltip ${!!message && 'tooltip-open'} tooltip-primary`} data-tip={message}>
        <Avatar status={status} src={image} alt={username} />
      </div>
      <div className="flex flex-col ml-4 space-y-2">
        <div className="truncate">{username}</div>
        <div>
          🍅 ×<span>{score}</span>
        </div>
      </div>
    </div>
  );
};

export default User;
