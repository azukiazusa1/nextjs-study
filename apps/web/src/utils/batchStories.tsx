import { composeStories } from '@storybook/testing-react';
import { StoryFile } from '@storybook/testing-react/dist/types';
import { render } from '@testing-library/react';
import React from 'react';

export const batchStories = (stories: StoryFile) => {
  const testCases = Object.values(composeStories(stories)).map((Story: any) => [
    Story.storyName,
    Story,
  ]);

  test.each(testCases)('Renders %s story', async (_storyName, Story) => {
    const tree = await render(<Story />);
    expect(tree.baseElement).toMatchSnapshot();
  });
};
