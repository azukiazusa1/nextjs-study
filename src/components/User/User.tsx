import React from 'react';

import Avatar from '@/components/Avatar/Avatar';

type Props = {
  className?: string;
  username: string;
  image: string;
  status: 'online' | 'offline' | '';
  quantity: number;
};

const User: React.FC<Props> = ({ className, username, image, status, quantity }: Props) => {
  return (
    <div className={`flex ${className}`}>
      <Avatar status={status} src={image} alt={username} />
      <div className="flex flex-col ml-4">
        <div>{username}</div>
        <div>🍅 × {quantity}</div>
      </div>
    </div>
  );
};

export default User;
