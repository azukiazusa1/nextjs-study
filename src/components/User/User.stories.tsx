import { ComponentMeta, ComponentStory } from '@storybook/react';
import React from 'react';

import User from './User';

export default {
  component: User,
} as ComponentMeta<typeof User>;

const Template: ComponentStory<typeof User> = (args) => <User {...args} />;

export const Default = Template.bind({});
Default.args = {
  username: '@johndoe',
  image:
    'https://images.unsplash.com/photo-1584361853901-dd1904bb7987?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1974&q=80',
  status: 'online',
  quantity: 12,
};

export const isOffline = Template.bind({});
isOffline.args = {
  username: '@johndoe',
  image:
    'https://images.unsplash.com/photo-1584361853901-dd1904bb7987?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1974&q=80',
  status: 'offline',
  quantity: 12,
};
