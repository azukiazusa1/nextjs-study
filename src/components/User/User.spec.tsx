import { batchStories } from '@/utils/batchStories';

import * as stories from './User.stories';

describe('components/User', () => {
  batchStories(stories);
});
