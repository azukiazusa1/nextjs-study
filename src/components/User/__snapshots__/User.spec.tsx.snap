// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/User Renders Default story 1`] = `
<body>
  <div>
    <div
      class="flex undefined"
    >
      <div
        class="avatar online undefined"
      >
        <div
          class="w-12 rounded-full"
        >
          <img
            alt="@johndoe"
            height="48px"
            src="https://images.unsplash.com/photo-1584361853901-dd1904bb7987?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1974&q=80"
            width="48px"
          />
        </div>
      </div>
      <div
        class="flex flex-col ml-4"
      >
        <div>
          @johndoe
        </div>
        <div>
          üçÖ √ó 
          12
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`components/User Renders isOffline story 1`] = `
<body>
  <div>
    <div
      class="flex undefined"
    >
      <div
        class="avatar offline undefined"
      >
        <div
          class="w-12 rounded-full"
        >
          <img
            alt="@johndoe"
            height="48px"
            src="https://images.unsplash.com/photo-1584361853901-dd1904bb7987?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1974&q=80"
            width="48px"
          />
        </div>
      </div>
      <div
        class="flex flex-col ml-4"
      >
        <div>
          @johndoe
        </div>
        <div>
          üçÖ √ó 
          12
        </div>
      </div>
    </div>
  </div>
</body>
`;
